reflex-dom-core/src/Reflex/Dom/WebSocket.hs

```diff
@@ -35,19 +36,38 @@ import Reflex.TriggerEvent.Class
 
 import Control.Concurrent
 import Control.Concurrent.STM
-import Control.Exception (SomeException)
+import Control.Exception
 import Control.Lens
 import Control.Monad hiding (forM, forM_, mapM, mapM_, sequence)
 import Control.Monad.IO.Class
 import Control.Monad.State
+import Data.Aeson hiding (json)
 import Data.ByteString (ByteString)
 import Data.Default
 import Data.IORef
+import Data.JSString (JSString)
 import Data.Maybe (isJust)
 import Data.Text
+import Data.Text.Encoding
 import GHCJS.DOM.Types (runJSM, askJSM, MonadJSM, liftJSM, JSM)
+import GHCJS.Marshal
+import qualified Language.Javascript.JSaddle as JS (jsg, js1)
 import qualified Language.Javascript.JSaddle.Monad as JS (catch)
 
+#ifdef ghcjs_HOST_OS
```

Comment:
How hard would it be to split the GHC and GHCJS stuff into /src-ghc and /src-ghcjs instead of using CPP?